### ISSUE TITLE
[Feature request]

### ISSUE THREAD

**Issue by cactuspeeler** [leoetlino/project-restoration#255]
Hi!
I hope you get around to reading this. I was wondering if maybe you would be willing to put in link rolling when jumping from a higher place (while holding the control stick forward) like from the n64 version. Its really bothered me since oot3d didnt have it either. Link just kind of lands straight on his feet. 
I dont know if its hard to do but youve done an amazing job already with the project. I would really appreciate it!
Kind regards,
CactusPeeler

**Comment by DerBaumBoy** [comment]
OoT3D does have this, you have to hold the circle pad forward and press roll with the correct timing when hitting the ground. It is pretty hard to hit the timing tho. I will test if MM3D does already have this feature as soon as I am able to. If it does not I will take a shot at implementing, although I cant promise anything.

**Comment by Licuatronix** [comment]
> OoT3D does have this, you have to hold the circle pad forward and press roll with the correct timing when hitting the ground. It is pretty hard to hit the timing tho. I will test if MM3D does already have this feature as soon as I am able to. If it does not I will take a shot at implementing, although I cant promise anything.

Yeah, no thats not how it works, you see the land roll exists in oot3d but someone in grezzo messed up a single code line that was important and ended up breaking it

**Comment by KingLuigi128** [comment]
For anyone curious about this and would like some more details,
there was a video just last month that looked into this issue for OoT3D.
https://www.youtube.com/watch?v=GIlzJQxGs54

I imagine that same code was carried over to MM3D with little to no changes.

**Comment by Licuatronix** [comment]
I actually used that video and a oot3d Randomizer fork for make a quick code that actually fixed the roll
https://github.com/Licuatronix/OoT3D_Randomizer/blob/main/code/src/actors/player.c
if we have luck the same can be done to mm3d

**Comment by cactuspeeler** [comment]
> I actually used that video and a oot3d Randomizer fork for make a quick code that actually fixed the roll
> https://github.com/Licuatronix/OoT3D_Randomizer/blob/main/code/src/actors/player.c
> if we have luck the same can be done to mm3d

Can you send me a patch file for my 3ds?

**Comment by Licuatronix** [comment]
you can just download the only release from my fork, sinse it is the randomizer you can easily apply the vanilla preset or just play with randomizer but the "land fix" will always be there
